server/sonar-server/src/main/java/org/sonar/server/webhook/ws/WebhookDeliveriesAction.java

```diff
@@ -0,0 +1,156 @@
+import static java.util.Objects.requireNonNull;
+import static org.sonar.api.utils.DateUtils.formatDateTime;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+
+public class WebhookDeliveriesAction implements WebhooksWsAction {
+
+  private static final String COMPONENT_PARAM = "componentKey";
+  private static final String TASK_PARAM = "ceTaskId";
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final ComponentFinder componentFinder;
+
+  public WebhookDeliveriesAction(DbClient dbClient, UserSession userSession, ComponentFinder componentFinder) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.componentFinder = componentFinder;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction action = controller.createAction("deliveries")
+      .setSince("6.2")
+      .setDescription("Get the recent deliveries for a specified project or Compute Engine task.<br/>" +
+        "Require 'Administer System' permission.<br/>" +
+        "Note that additional information are returned by api/webhooks/delivery.")
+      .setResponseExample(Resources.getResource(this.getClass(), "example-deliveries.json"))
+      .setHandler(this);
+
+    action.createParam(COMPONENT_PARAM)
+      .setDescription("Key of the project")
+      .setExampleValue("my-project");
+
+    action.createParam(TASK_PARAM)
+      .setDescription("Id of the Compute Engine task")
+      .setExampleValue(Uuids.UUID_EXAMPLE_01);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    // fail-fast if not logged in
+    userSession.checkLoggedIn();
+
+    String ceTaskId = request.param(TASK_PARAM);
+    String componentKey = request.param(COMPONENT_PARAM);
+    checkArgument(ceTaskId != null ^ componentKey != null, "Either parameter '%s' or '%s' must be defined", TASK_PARAM, COMPONENT_PARAM);
+
+    Data data = loadFromDatabase(ceTaskId, componentKey);
+    data.ensureAdminPermission(userSession);
+    data.writeTo(request, response);
+  }
+
+  private Data loadFromDatabase(@Nullable String ceTaskId, @Nullable String componentKey) {
+    ComponentDto component = null;
+    List<WebhookDeliveryLiteDto> deliveries;
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      if (componentKey != null) {
+        component = componentFinder.getByKey(dbSession, componentKey);
+        deliveries = dbClient.webhookDeliveryDao().selectOrderedByComponentUuid(dbSession, component.uuid());
+      } else {
+        deliveries = dbClient.webhookDeliveryDao().selectOrderedByCeTaskUuid(dbSession, ceTaskId);
+        Optional<String> deliveredComponentUuid = deliveries
+          .stream()
+          .map(WebhookDeliveryLiteDto::getComponentUuid)
+          .findFirst();
+        if (deliveredComponentUuid.isPresent()) {
+          component = componentFinder.getByUuid(dbSession, deliveredComponentUuid.get());
```

Comment:
`getByUuid` will raise a error "Component id '%s' not found" when requesting the deliveries of a task of a deleted component. Granted, this should not occur if data is consistent and purge is working ok.
But, if the error occurs, since no component has been provided in the request, the error will be hard to understand by the user.